backend/src/test/java/gov/cdc/usds/simplereport/validators/PatientBulkUploadFileValidatorTest.java

```diff
@@ -0,0 +1,157 @@
+
+  @Test
+  void validFile() {
+    // GIVEN
+    InputStream input = loadCsv("patientBulkUpload/valid.csv");
+    // WHEN
+    List<FeedbackMessage> errors = patientBulkUploadFileValidator.validate(input);
+    // THEN
+    assertThat(errors).isEmpty();
+  }
+
+  @Test
+  void missingHeaders_returnsErrors() {
+    // GIVEN
+    InputStream input = loadCsv("patientBulkUpload/missingHeaders.csv");
+    // WHEN
+    List<FeedbackMessage> errors = patientBulkUploadFileValidator.validate(input);
+    // THEN
+    assertThat(errors).hasSize(12);
+    List<String> errorMessages =
+        errors.stream().map(FeedbackMessage::getMessage).collect(Collectors.toList());
+    assertThat(errorMessages)
+        .contains(
+            "first_name is a required column.",
+            "last_name is a required column.",
+            "race is a required column.",
+            "date_of_birth is a required column.",
+            "biological_sex is a required column.",
+            "ethnicity is a required column.",
+            "street is a required column.",
+            "state is a required column.",
+            "zip_code is a required column.",
+            "phone_number is a required column.",
+            "employed_in_healthcare is a required column.",
+            "resident_congregate_setting is a required column.");
+  }
+
+  @Test
+  void missingOptionalFields_isValid() {
+    // GIVEN
+    InputStream input = loadCsv("patientBulkUpload/missingOptionalFields.csv");
+    // WHEN
+    List<FeedbackMessage> errors = patientBulkUploadFileValidator.validate(input);
+    // THEN
+    assertThat(errors).isEmpty();
+  }
+
+  @Test
+  void missingRequiredFields_returnsErrors() {
+    // GIVEN
+    InputStream input = loadCsv("patientBulkUpload/missingRequiredFields.csv");
+    // WHEN
+    List<FeedbackMessage> errors = patientBulkUploadFileValidator.validate(input);
+    // THEN
+    assertThat(errors).hasSize(2);
+    List<String> errorMessages =
+        errors.stream().map(FeedbackMessage::getMessage).collect(Collectors.toList());
+    assertThat(errorMessages)
+        .contains("race is a required column.", "ethnicity is a required column.");
+  }
+
+  @Test
+  void extraColumns_isValid() {
+    // GIVEN
+    InputStream input = loadCsv("patientBulkUpload/extraColumns.csv");
+    // WHEN
+    List<FeedbackMessage> errors = patientBulkUploadFileValidator.validate(input);
+    // THEN
+    assertThat(errors).isEmpty();
+  }
+
+  @Test
+  void invalidValues_returnError() {
+    // GIVEN
+    InputStream input = loadCsv("patientBulkUpload/invalidValues.csv");
+    // WHEN
+    List<FeedbackMessage> errors = patientBulkUploadFileValidator.validate(input);
+    // THEN
+    assertThat(errors).hasSize(12);
+    List<String> errorMessages =
+        errors.stream().map(FeedbackMessage::getMessage).collect(Collectors.toList());
+    assertThat(errorMessages)
+        .contains(
+            "11/3/80 is not a valid value for column date_of_birth",
```

Comment:
Another area where I thought for a while about the best way to test these invalid values. I ultimately landed on one large bad file, vs several smaller files that are individually tested like we had before ("invalidZipCode.csv", "invalidDateOfBirth.csv", etc) because one file is more easily maintainable if we ever add or remove columns from the schema. The downside is it makes the test less isolated.

Reply:
I think it's ok, given the one large bad file covers all the possible invalid value errors. I think it's pretty clear reading the test what's being tested.